sklearn/ensemble/_stacking.py

```diff
@@ -0,0 +1,668 @@
+            of the classifiers as well.
+        """
+        return super()._get_params('estimators', deep=deep)
+
+    def _concatenate_predictions(self, predictions):
+        """Concatenate the predictions of each first layer learner.
+
+        This helper is in charge of ensuring the preditions are 2D arrays and
+        it will drop one of the probability column when using probabilities.
+        Indeed, the p(y|c=0) = 1 - sum_{k=1}^{K} p(y|c=k)
+        """
+        X_meta = []
+        for est_idx, preds in enumerate(predictions):
+            # case where the the estimator returned a 1D array
+            if preds.ndim == 1:
+                X_meta.append(preds.reshape(-1, 1))
+            else:
+                if self.stack_method_[est_idx] == 'predict_proba':
+                    # Remove the first column when using probabilities. It is
+                    # collinear with others.
+                    X_meta.append(preds[:, 1:])
+                else:
+                    X_meta.append(preds)
+        return np.concatenate(X_meta, axis=1)
+
+    @staticmethod
+    def _method_name(name, estimator, method):
+        if estimator == 'drop':
+            return None
+        if method == 'auto':
+            if getattr(estimator, 'predict_proba', None):
+                return 'predict_proba'
+            elif getattr(estimator, 'decision_function', None):
+                return 'decision_function'
+            else:
+                return 'predict'
+        else:
+            if not hasattr(estimator, method):
+                raise ValueError('Underlying estimator {} does not implement '
+                                 'the method {}.'.format(name, method))
+            return method
+
+    def fit(self, X, y, sample_weight=None):
+        """Fit the estimators.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix} of shape (n_samples, n_features)
+            Training vectors, where `n_samples` is the number of samples and
+            `n_features` is the number of features.
+
+        y : array-like of shape (n_samples,)
+            Target values.
+
+        sample_weight : array-like of shape (n_samples,) or None
+            Sample weights. If None, then samples are equally weighted.
+            Note that this is supported only if all underlying estimators
+            support sample weights.
+
+        Returns
+        -------
+        self : object
+        """
+        names, estimators_ = self._validate_estimators()
+        self._validate_final_estimator()
+
+        has_estimator = False
+        for est in estimators_:
+            if est != 'drop':
+                has_estimator = True
+
+        if not has_estimator:
+            raise ValueError(
+                "All estimators are dropped. At least one is required "
+                "to be an estimator."
+            )
+
+        stack_method = [self.stack_method] * len(estimators_)
```

Comment:
maybe we don't need to construct a list?

Reply:
This is annoying otherwise for the later construction of `stack_method_`